### PR TITLE
robots: introduces robots.txt

### LINKED ISSUE #700: system: introduce `robots.txt`
A sensible `robots.txt` should be introduced.

### PR DESCRIPTION
**Pull request by ioannistsanaktsidis**
* Closes #700.

Signed-off-by: Ioannis Tsanaktsidis <ioannis.tsanaktsidis@cern.ch>

### CODE REVIEW

File: robots.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+User-agent: *
+Allow: /
```
Comment (tiborsimko) [inline robots.txt]: There is no `Allow` field in the robots specs as per http://www.robotstxt.org/robotstxt.html ... 

Let's see whether we have some set of pages to restrict; that would be the cleanest solution to the no-allow problem.  

Maybe `/api/...` URLs, provided that we expose things via `/record/NNN/export/...` and `/record/NNN/files/FFF` kind of URLs?

Comment (tiborsimko) [reply robots.txt]: Depending on the #2247, yes, I'd say let's disallow `/api` for now.

**Comment by tiborsimko** [comment]
We also need to plug this to the app route map, so that `opendata.cern.ch/robots.txt` would reply :wink: Please move the file somewhere under `static` maybe and please define a new view endpoint that would serve it.

**Comment by tiborsimko** [comment]
(1) Looks good, but the file is exported as if it were HTML, not as plain text:

```console
$ wget -q -S --spider http://0.0.0.0:5000/robots.txt
  HTTP/1.0 200 OK
  Content-Type: text/html; charset=utf-8
  Content-Length: 29
  Server: Werkzeug/0.12.2 Python/2.7.5
  Date: Mon, 09 Apr 2018 08:26:43 GMT
```

Can you please change the content type?

(2) Also, I've seen that one can expose sitemap via `robots.txt`: https://www.sitemaps.org/protocol.html#submit_robots

```
Sitemap: http://www.example.com/sitemap.xml
```

Let's perhaps add that?  Although beware of QA, DEV, and PROD base URLs... we'd have to do that from within Pythonic app then.

**Comment by ioannistsanaktsidis** [comment]
@tiborsimko so should I remove the old implementation and add the sitemap one ?

**Comment by tiborsimko** [comment]
The old implementation is OK, we just need one more line with the `Sitemap` line (and fix the content type of the output). You are right, we can hard-code the PROD URL `http://opendata.cern.ch/`, no need to go into Pythonic app.

Regarding for QA and DEV, we should forbid everything... I can amend while merging up, We don't want these two development sites to be crawled and indexed.

**Comment by ioannistsanaktsidis** [comment]
@tiborsimko everything is updated :)

**Pull request merged**